### ISSUE TITLE
No kubernetes versions displayed after factory reset on windows

### ISSUE THREAD

**Issue by AndiBoegelein** [rancher-sandbox/rancher-desktop#7615]
### Actual Behavior

After a factory reset on first startup of rancher desktop or under Preferences -> Kubernetes there are no kubernetes Versions displayed.

![image](https://github.com/user-attachments/assets/ac40085f-5ec1-41a8-8006-c2a4cd928a95)

### Steps to Reproduce

Factory Reset RD
Restart RD
Try to select kubernetes version

### Result

No kubernetes versions are displayed

### Expected Behavior

Display all working/available kubernetes versions as it was before factory reset

### Additional Information

I attached most of the logs that have any information in them.
Relevant logs seem to be background.log with the hint that the request to https://desktop.version.rancher.io/v1/checkupgrade failed,
Also important to note is the wsl setup for certificates in the wsl-exec.log
[wsl-init.log](https://github.com/user-attachments/files/17325399/wsl-init.log)
[background.log](https://github.com/user-attachments/files/17325400/background.log)
[rancher-desktop-guestagent.log](https://github.com/user-attachments/files/17325401/rancher-desktop-guestagent.log)
[wsl-exec.log](https://github.com/user-attachments/files/17325402/wsl-exec.log)
[buildkitd.log](https://github.com/user-attachments/files/17325403/buildkitd.log)
[vm-switch.log](https://github.com/user-attachments/files/17325404/vm-switch.log)
[network-setup.log](https://github.com/user-attachments/files/17325405/network-setup.log)

### Rancher Desktop Version

1.16.0

### Rancher Desktop K8s Version

deactivated because no version is offered

### Which container engine are you using?

containerd (nerdctl)

### What operating system are you using?

Windows

### Operating System / Build Version

Windows 10 Enterprise

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

None

### Windows User Only

I am using a coporate proxy with authentication to access certain internet resources

**Comment by mook-as** [comment]
Hmm, the logs say:
> FetchError: request to https://update.k3s.io/v1-release/channels failed, reason: self signed certificate in certificate chain

Are you using any proxies (or possibly corporate transparent proxies)? Though we should have code to deal with that…

**Comment by angolding** [comment]
I am running into this issue and I am behind a proxy

**Comment by AndiBoegelein** [comment]
> Hmm, the logs say:
> 
> > FetchError: request to https://update.k3s.io/v1-release/channels failed, reason: self signed certificate in certificate chain
> 
> Are you using any proxies (or possibly corporate transparent proxies)? Though we should have code to deal with that…

Yes I am behind a corporate proxy. I already used the RD Feature to enable proxy use (with authentication) but it does not seem to have any effect on loading kubernetes versions.

**Comment by mook-as** [comment]
Right, the proxy settings are for the VM once that's up; but the issue currently is that things being fetched from the host (via JavaScript) isn't picking up the host CA certificates for some reason.  We'll need to look at that code again.

**Comment by AndiBoegelein** [comment]
> Right, the proxy settings are for the VM once that's up; but the issue currently is that things being fetched from the host (via JavaScript) isn't picking up the host CA certificates for some reason. We'll need to look at that code again.

Hi @mook-as thanks for giving the outlook of this working in a future version.
Just FYI I remebered that I used an older version before I used factory reset.
In this older version everything seemed to work fine. To test this I just installed the older version (1.12.3) and tested if it would still work. Lo and behold it is working with the older version without changing anything else (system wise):
![image](https://github.com/user-attachments/assets/34264e62-00a6-49e3-abd7-bc947b0b5535)


Maybe this will help you identify the bug comparing the code to an older version.

**Comment by mook-as** [comment]
This should be resolved in v1.17 thanks to #7909. Please let us know if that isn't the case, thanks!
